Neos.Neos/Classes/Service/XliffService.php

```diff
@@ -111,22 +111,39 @@ public function getCachedJson(Locale $locale): string
                     $file = $this->xliffFileProvider->getFile($fileId, $locale);
 
                     foreach ($file->getTranslationUnits() as $key => $value) {
-                        $valueToStore = !empty($value[0]['target']) ? $value[0]['target'] : $value[0]['source'];
-                        if ($this->scrambleTranslatedLabels) {
-                            $valueToStore = str_repeat('#', UnicodeFunctions::strlen($valueToStore));
-                        }
+                        $valueToStore = $this->getTranslationUnitValue($value);
+                        $valueToStore = count($valueToStore) > 1 ? $valueToStore : array_shift($valueToStore);
                         $this->setArrayDataValue($labels, str_replace('.', '_', $packageKey) . '.' . str_replace('/', '_', $sourceName) . '.' . str_replace('.', '_', $key), $valueToStore);
                     }
                 }
             }
 
-            $json = json_encode($labels);
+            $json = json_encode($labels, true);
```

Comment:
I think you should use a constant for the options instead of true. Like described on http://www.php.net/manual/en/function.json-encode.php
Or what is the default behaviour of `true` here?

Reply:
Ah this is a relict I forgot to remove. Will be removed soonish ... good catch.